user-operator/src/main/java/io/strimzi/operator/user/model/KafkaUserModel.java

```diff
@@ -325,7 +325,7 @@ public String getUserName()    {
         } else if (isScramUser()) {
             return getScramUserName(name);
         } else {
-            throw new RuntimeException("At least one authentication mechanism has to be selected");
+            return getName();
```

Comment:
in this case are you showing the name of the resource, why If it's not TLS or SCRAM SHA user?

Reply:
This is when there is no authentication section at all. This can happen for example when the user configures only ACLs for an OAuth user. In such case we take the resource name as the username.